### PR TITLE
Rate limiting post

### PR DESCRIPTION
**Pull request by jkemp101**

### CODE REVIEW

File: _posts/2020-03-12-rate-limiting-at-close.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+to make sure the TTL is reliably set once so the key expires and resets the
+counter to `0`. This can be guaranteed by placing the `incr` and `expire`
+commands in the same script like this:
+
+    local current_count
+    current_count = redis.call("incr",KEYS[1])
+    if tonumber(current_count) == 1 then
+        redis.call("expire",KEYS[1],ARGV[1])
+    end
+    return current_count
```
Comment (philfreo) [inline _posts/2020-03-12-rate-limiting-at-close.md]: Does this get syntax highlighting?
It doesn't in GitHub.
You might have better luck with triple backticks and the word `lua`

Comment (jkemp101) [reply _posts/2020-03-12-rate-limiting-at-close.md]: Yep that change added syntax highlighting

File: _posts/2020-03-12-rate-limiting-at-close.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+that dynamically adjusts the RPS throttle knob, taking into account ES load,
+Mongo load, and the reindexer queue length. This solved a few problems:
+
+- Automatically throttling up in the evenings and weekends when normal loads drop off
+- Automatically throttling down
+    - When other higher priority workloads require compute resources
+    - When reindexing documents that are more difficult to index
+
+Calibrating the Governor was quite a challenge (there are literally [books
+written](https://www.amazon.com/s?k=pid+controller+tuning&i=stripbooks&ref=nb_sb_noss)
```
Comment (tsx) [inline _posts/2020-03-12-rate-limiting-at-close.md]: This has to be an affiliate link that earns some extra money :rofl:

File: _posts/2020-03-12-rate-limiting-at-close.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+notifications for the same calendar. To protect the underlying MySQL database we
+added a LimitLion throttle to the existing [Nylas
+endpoint](https://github.com/closeio/nylas/blob/082b18bfed00e28335084d5c8162336e3f33a719/inbox/webhooks/gpush_notifications.py#L85-L105)
+that handles these notifications. The `limitlion.throttle()` call with a
+`rps=.5` using the default window of 5 seconds allows up to 3 notifications to
+be processed every 5 seconds for each calendar. Anything beyond that is ignored.
+
+    @app.route('/calendar_update/<calendar_public_id>', methods=['POST'])
+    def event_update(calendar_public_id):
+        request.environ\['log_context'\]['calendar_public_id'] = calendar_public_id
```
Comment (thomasst) [inline _posts/2020-03-12-rate-limiting-at-close.md]: Should remove these backslashes. They are showing up in the rendered version.

**Pull request merged**